sql3/planner/expression.go

```diff
@@ -1998,6 +1998,48 @@ func (n *boolLiteralPlanExpression) WithChildren(children ...types.PlanExpressio
 	return n, nil
 }
 
+// represents date system variable such as CURRENT_DATE and CURRENT_DATETIME
+type dateSysVariablePlanExpression struct {
```

Comment:
Do we need a variable expression type this specific? We could have a single variable expression type and then switch on name...